packages/core/src/components/HelpDrawer/HelpDrawer.jsx

```diff
@@ -48,12 +48,16 @@ export class HelpDrawer extends React.PureComponent {
             </Button>
           </div>
         </div>
-        <div className="ds-c-help-drawer__body ds-u-md-font-size--small ds-u-lg-font-size--base ds-u-padding--2">
-          {children}
-        </div>
-        <div className="ds-c-help-drawer__footer ds-u-fill--primary-alt-lightest ds-u-md-font-size--small ds-u-lg-font-size--base ds-u-padding--2">
-          <h4 className="ds-text ds-u-margin--0">{footerTitle}</h4>
-          <div className="ds-text ds-u-margin--0">{footerBody}</div>
+        <div className="ds-c-help-drawer__body">
```

Comment:
@davidakennedy @line47
I referencing [usa spending's glossary](https://www.usaspending.gov/#/) during this PR, and I noticed their markup used `aside`, `role: dialog`, and even a div with `role: navigation` wrapping the close button. 
https://gyazo.com/4caf6f1a806eb67a77d5f4d1ab58d939

Without researching too much, some of these sound like they should apply to us, i.e. I could see our help drawer using `aside`, `header`, and `footer`. I also need to do more screen reader testing for the roles.

What do yall think, should we update our markup to be more semantic as well? And if so, what should we change?

Reply:
> Without researching too much, some of these sound like they should apply to us, i.e. I could see our help drawer using aside, header, and footer. I also need to do more screen reader testing for the roles.
> 
> What do yall think, should we update our markup to be more semantic as well? And if so, what should we change?

This is a good question to ask. Here are some of my thoughts:

As currently implemented, I see the help drawer more like a [nonmodal dialog](https://www.nngroup.com/articles/modal-nonmodal-dialog/), maybe even closer to an accordion in functionality, but not appearance.

The spending glossary is a bit different in that they trap focus, like a true modal, even if the visual design is more like a nonmodal. It's also more complex than our help drawer in that when you click buttons in it, it slides to the word definition. Our drawer is usually used in less complex ways.

In most of the ways I've seen the help drawer used, it's less about needing to grab people's attention with a must-do task and more about read this important information. So I don't think the `role="modal"` markup makes sense for us. Unless people are using it in more complex ways. But maybe that means we need a separate component. That role cancels out the normal role of an `<aside>` element, which is complementary. So using an `<aside>` in the spending glossary is kind of a wash. If we used an `<aside>` in the help drawer, we get [this warning in aXe](https://dequeuniversity.com/rules/axe/3.3/landmark-complementary-is-top-level). So I wouldn't do that. I do think `<header>` and `<footer>` would help, but only if the content is log enough (to be syndicated, like an article), and the header and footer elements had other elements in them. Like dates, categories, etc. I probably wouldn't add them now.

I do think we could improve the help drawer button semantics with `aria-pressed` or `aria-expanded and aria-controls`. That would give the control pressed to open the drawer more meaning.

Reply:
Gotcha, thank you for looking into this!

My question is, is there any benefit to adding an aside, and do you think that accurately describes the help drawer? I think we might be able to fix that aXe error if there’s some benefit to using aside

And as far as header/footer, I believe the header is always pretty simple (title and a button for closing), and footers on App3 include some links and phone numbers for additional support. Also, help drawer content can range from a paragraph, to a lot of text with lists and links. The main reason why we wouldn’t use header/footer in this case is because most use cases aren’t complex enough to warrant this right?

Reply:
I was thinking about `aria-expanded/aria-controls` and `aria-pressed`.  I wonder if it makes sense to add `aria-expanded/aria-controls` considering you will only see the button if `aria-expanded` is true. In my mind expanded/controls is more suited for a button that is always visible, but toggles an expandable section. I feel similarly with `aria-pressed`, which in my understanding is for simulating a "toggle button" for screen readers, which doesnt seem as applicable in this situation.

Reply:
> is there any benefit to adding an aside, and do you think that accurately describes the help drawer?

There's not a strong benefit to adding one from an assistive technology perspective. Screen reader users will be able to navigate/skip to it because it's a landmark, but because it would be nested in another landmark (most likely), screen reader users won't be able to skip over it entirely. It goes against a best practice, but not the end of the world either. This is one of those cases that's hard because we won't always know the markup around the component.

I do think an aside is the right element for this type of content, at least how I've seen the help drawer used in the wild.

> The main reason why we wouldn’t use header/footer in this case is because most use cases aren’t complex enough to warrant this right?

Right. We could maybe make that an option with the ability to add additional markup, if needed? I could see it being needed in some cases for sure.

Re: `aria-expanded/aria-controls` and `aria-pressed`, this is a tough case because the component is somewhere between an accordion and modal. When I was thinking about `aria-expanded/aria-controls` and `aria-pressed`, I was thinking of adding it to the button that triggered opening the help drawer. That way, a person would know that the button is going to expand something when clicked.

Reply:
Ah those explanations help a lot thank you! It sounds like for now its probably not a big priority to add `aside` and `header/footer` than. 

Re: `aria-expanded/aria-controls` and `aria-pressed`, I see what you mean about putting it on the button that triggers opening, that makes sense. I feel like the fact that the open and close buttons are separate is slightly more similar to a modal, so its definitely tricky. 

This was really good discussion and context, thank you @davidakennedy again! My understanding is that theres no urgent a11y changes that this PR should wait on.